src/push.rs

```diff
@@ -99,12 +99,46 @@ pub async fn push_profile(data: PushProfileData<'_>) -> Result<(), PushProfileEr
         Command::new("nix-build")
     };
 
+    let store_address = format!("ssh-ng://{}@{}",
+        if data.deploy_data.profile.generic_settings.ssh_user.is_some() {
+            &data.deploy_data.profile.generic_settings.ssh_user.as_ref().unwrap()
+        } else {
+            &data.deploy_defs.ssh_user
+        },
+        data.deploy_data.node.node_settings.hostname
+    );
+
     if data.supports_flakes {
         build_command.arg("build").arg(derivation_name)
     } else {
         build_command.arg(derivation_name)
     };
 
+    // copy derivation over to remote server
+    if data.supports_flakes && data.deploy_data.merged_settings.remote_build.unwrap_or(false) {
+        info!("Building on remote host");
+
+        // copy the derivation to remote host so it can be built there
+        let copy_command_status = Command::new("nix").arg("copy")
+            .arg("-s")  // fetch dependencies from substitures, not localhost
+            .arg("--to").arg(&store_address)
```

Comment:
Looks like this doesn't inherit an ssh port, thus an attempt to use it may fail, e.g.
```
ssh: connect to host skat.aquarius.serokell.team port 22: Network is unreachable
error: cannot open a connection to remote store 'ssh-ng://rvem@<bla-bla>': error: unexpected end-of-file
```
While deploy configuration uses non-default ssh port